src/dune_rules/preprocess.ml

```diff
@@ -12,9 +12,13 @@ module Pps_and_flags = struct
     and+ syntax_version = Dune_lang.Syntax.get_exn Stanza.syntax in
     let pps, more_flags =
       List.partition_map l ~f:(fun s ->
-        match String_with_vars.is_prefix ~prefix:"-" s with
-        | Yes -> Right s
-        | No | Unknown _ ->
+        match
+          ( String_with_vars.is_prefix ~prefix:"-" s
+          , String_with_vars.is_prefix ~prefix:"+" s )
+        with
+        | Yes, _ -> Right s
+        | _, Yes when syntax_version >= (3, 18) -> Right s
+        | (No | Unknown _), _ ->
```

Comment:
I think you could improve the error message here with a hint for `prefix = "+"`  when `syntax_version < (3, 18)`